dot/types/inherents.go

```diff
@@ -5,23 +5,42 @@ package types
 
 import (
 	"bytes"
-	"encoding/binary"
-	"errors"
 	"fmt"
 	"math/big"
 
 	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
-var (
-	// Timstap0 is an inherent key.
-	Timstap0 = []byte("timstap0")
-	// Babeslot is an inherent key.
-	Babeslot = []byte("babeslot")
-	// Uncles00 is an inherent key.
-	Uncles00 = []byte("uncles00")
+// InherentIdentifier is an identifier for an inherent.
+type InherentIdentifier uint
+
+const (
+	// Timstap0 is the identifier for the `timestamp` inherent.
+	Timstap0 InherentIdentifier = iota
+	// Babeslot is the BABE inherent identifier.
+	Babeslot
+	// Uncles00 is the identifier for the `uncles` inherent.
+	Uncles00
```

Comment:
why not instead have these as `string` and then just use `[]byte(Timstap0)` where needed? Seems better than iota + method.

Also probably too out of scope, but these are not types, maybe they can be moved to lib/babe or lib/babe/constants (in case of cycle dependency) 🤔

Reply:
its not `[]byte(Timstap0)`, its [8]byte. It save me from doing things like `kb := [8]byte{}; copy(kb[:], key)`

> Also probably too out of scope, but these are not types, maybe they can be moved to lib/babe or lib/babe/constants (in case of cycle dependency) thinking

`InherentIdentifier` should stay where InherentsData stays. We could however move inherents.go file to `lib/babe`. It makes more sense there anyway.

Reply:
Inherents get used in lib/runtime, and lib/sync as well. Let's just keep them here in dot/types. This saves possible cyclical dependencies